### ISSUE TITLE
Cannot pull from Docker Registry on other node

### ISSUE THREAD

**Issue by lorenzvth7** [openshift/openshift-ansible#632]
I have one master-server, a node1-server and a node2-server. They're all on CentOS instances of AWS. I've created my docker registry on my master-machine. After a while I saw the registry was put on node 1. I don't know if this is normal. After that I want to create my first project and app. Also on the master-machine but it was all in containers on my node1. When I tried to create more pods (than one) I did the following command: 
```
oc scale rc test-1 --replicas=4
```
After that the're were created 4 pods. 2 of the pods are on the node1 (where the registry is), the other 2 on node2. The pods on node1 are running fine but the pods on node2 are unable to find the image.

```
[root@ip-10-0-0-135 centos]# oc get pods
NAME           READY     STATUS          RESTARTS   AGE
test-1-6oz3v   1/1       Running         0          1h
test-1-8zt9q   1/1       Running         0          1h
test-1-build   0/1       ExitCode:0      0          1h
test-1-fkvas   0/1       no such image   0          1h
test-1-or31p   0/1       API error (500): invalid registry endpoint "http://172.30.26.202:5000/v0/". HTTPS attempt: unable to ping registry endpoint https://172.30.26.202:5000/v0/
v2 ping attempt failed with error: Get https://172.30.26.202:5000/v2/: read tcp 172.30.26.202:5000: connection reset by peer
 v1 ping attempt failed with error: Get https://172.30.26.202:5000/v1/_ping: read tcp 172.30.26.202:5000: connection reset by peer. HTTP attempt: unable to ping registry endpoint http://172.30.26.202:5000/v0/
v2 ping attempt failed with error: Get http://172.30.26.202:5000/v2/: read tcp 172.30.26.202:5000: connection reset by peer
 v1 ping attempt failed with error: Get http://172.30.26.202:5000/v1/_ping: read tcp 172.30.26.202:5000: connection reset by peer
```
```
NAME      DOCKER REPO                    TAGS      UPDATED
test      172.30.26.202:5000/test/test   latest    About an hour ago
```
I think they can not access the docker registry but I don't know why. Is there a solution for this?

**Comment by nak3** [comment]
Only node2 can't run any pods, so it looks like firewall or security group issue between the node2 and master. Could you please test this steps on node2?

Accessing the Registry (NOTE: please just check if you can work `docker login -u ...`)
https://docs.openshift.org/latest/install_config/install/docker_registry.html#access

**Comment by lorenzvth7** [comment]
```docker login -u test-admin -e my@mail.com -p e41ZCpD40wATrG4kveSS5PL2EU5x8vvbs1MVn1cH7zE 172.30.26.202:5000``` doesn't give any prompt back

**Comment by lorenzvth7** [comment]
timeout: ```Error response from daemon: invalid registry endpoint "http://172.30.26.202:5000/v0/". HTTPS attempt: unable to ping registry endpoint https://172.30.26.202:5000/v0/
v2 ping attempt failed with error: Get https://172.30.26.202:5000/v2/: read tcp 172.30.26.202:5000: i/o timeout```

**Comment by nak3** [comment]
Since node1->master is working, most probably node2's network setting caused the issue.
Did `4789(UDP)` port is opend on node2?

**Comment by lorenzvth7** [comment]
The port is closed for all my machines (master+node, node1, node2).  But is it normal that I create the docker registry on my master and that it's placed on another server (node1)

**Comment by nak3** [comment]
No, I don't think so. `4789(UDP)` is used by SDN https://access.redhat.com/solutions/1520653
You can also see.
~~~
[root@ose3-node1 ~]# netstat -lapun |grep 4789
udp        0      0 0.0.0.0:4789            0.0.0.0:*  
~~~

**Comment by abutcher** [comment]
@lorenzvth7 To deploy the registry to a specific server you can use a selector w/ node labels.

For example, you could label the master as infra:

```
oc label node <master hostname> region=infra
```

And then when creating your registry, pass `--selector='region=infra'`. Help for `oadm registry` explains:

```
--selector='': Selector used to filter nodes on deployment. Used to run registries on a specific set of nodes.
```

**Comment by lorenzvth7** [comment]
Thanks @abutcher, didn't know that. @nak3 ; The ports are now open on each server but I get the same error.

**Comment by lorenzvth7** [comment]
I've restarted every docker instance and every master/node. Still the same issue.

Iptables: everything is accepted from anywhere to anywhere.
```
iptables --list
Chain INPUT (policy ACCEPT)
target     prot opt source               destination         
ACCEPT     udp  --  anywhere             anywhere             multiport dports 4789 /* 001 vxlan incoming */
ACCEPT     all  --  anywhere             anywhere             /* traffic from docker for internet */
ACCEPT     all  --  anywhere             anywhere             state RELATED,ESTABLISHED
...
```
Using right IP and port.
```
oc get services
NAME              CLUSTER_IP      EXTERNAL_IP   PORT(S)    SELECTOR                  AGE
docker-registry   172.30.26.202   <none>        5000/TCP   docker-registry=default   1d
```

**Comment by lorenzvth7** [comment]
This is the error at this moment. No issue anymore about the flag.
```
Error response from daemon: invalid registry endpoint "http://172.30.26.202:5000/v0/". HTTPS attempt: unable to ping registry endpoint https://172.30.26.202:5000/v0/
v2 ping attempt failed with error: Get https://172.30.26.202:5000/v2/: read tcp 172.30.26.202:5000: connection reset by peer
 v1 ping attempt failed with error: Get https://172.30.26.202:5000/v1/_ping: read tcp 172.30.26.202:5000: connection reset by peer. HTTP attempt: unable to ping registry endpoint http://172.30.26.202:5000/v0/
v2 ping attempt failed with error: Get http://172.30.26.202:5000/v2/: read tcp 172.30.26.202:5000: connection reset by peer
 v1 ping attempt failed with error: Get http://172.30.26.202:5000/v1/_ping: read tcp 172.30.26.202:5000: connection reset by peer
```

**Comment by lorenzvth7** [comment]
On the machine where my docker registry is placed I get an authorization error. I don't know it's related
```
Error response from daemon: no successful auth challenge for http://172.30.26.202:5000/v2/ - errors: [basic auth attempt to http://172.30.26.202:5000/v2/ realm "openshift" failed with status: 401 Unauthorized]
```

**Comment by abutcher** [comment]
@lorenzvth7 Did you login on the system from where you pushed, ip-10-0-0-144 ?

**Comment by lorenzvth7** [comment]
It's not really clear for me what you want to say but I'll just give a little overview of what I did:
I have a master/node-instance, a node1-instance and a node2-instance (so 2 real nodes and one on the master server)
So first of all I've created my docker registry as user system:admin in the default project of openshift:
```
oadm registry --config=admin.kubeconfig \
    --credentials=openshift-registry.kubeconfig \
```
My registry is successfully created on my node1 (because I did not use the selector). 
After that I want to test if my registry is accessible for my other nodes.
So here is what I do https://docs.openshift.org/latest/install_config/install/docker_registry.html#access
'Accessing the registry'
I'm able to login, to generate the token, to pull. But when I have to push the image to the repo I get the error (described above)
